tools/system_libs.py

```diff
@@ -611,6 +599,13 @@ def get_cflags(self):
     if self.includes:
       cflags += ['-I' + utils.path_from_root(i) for i in self._inherit_list('includes')]
 
+    if self.deterministic_paths:
+      source_dir = utils.path_from_root()
+      if self.batch_inputs:
+        relative_source_dir = os.path.relpath(source_dir, build_dir)
+        cflags += [f'-ffile-prefix-map={relative_source_dir}={DETERMINISITIC_PREFIX}']
```

Comment:
Can you make these two lines unconditional and avoid the need for the new `self.batch_inputs`?

Reply:
Done

Reply:
Do you have any other comments?